project_key/models/project_project.py

```diff
@@ -13,7 +13,7 @@ class Project(models.Model):
         comodel_name="ir.sequence", string="Key Sequence", ondelete="restrict"
     )
 
-    key = fields.Char(string="Key", size=10, required=False, index=True, copy=False)
+    key = fields.Char(size=10, required=False, index=True, copy=False)
```

Comment:
Remove size=X attribute in Char fields, as it's no longer valid for restricting the size of the strings.

see: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0

Reply:
That’s strange because in Odoo 14, size is used on the attachment’s checksum : https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/ir_attachment.py#L393
In the database, the type of the column is `character varying(40)`.
(the url is also defined as size 1024 and the information is found on the database too)

That’s still the case with Odoo 15 and 16.

Reply:
You can see it here:  https://github.com/odoo/odoo/blob/14.0/odoo/fields.py#L1624

You can leave it for me, it can be removed in the next version.